### ISSUE TITLE
Make front-end templates more idiomatic

### ISSUE THREAD

**Issue by mehtaphysical** [near/create-near-app#810]
Right now the front-end templates used by `create-near-app` aren't very idiomatic. It would be nice to improve this by creating libraries for each front-end framework like https://www.npmjs.com/package/near-react-hooks.

This has two major advantages:

1. the developer experience is much better as the NEAR code fits in better with the chosen framework
2. it will be easier to incorporate NEAR into existing applications by using the framework specific library directly.

If this is of interest I'd be happy to update the current react template to use `near-react-hooks`. Also I can work on creating an angular service library and a vue plugin.

**Comment by chadoh** [comment]
What specifically strikes you as un-idiomatic about each? 

I've used React extensively and currently find our React setup straightforward and readable. I have found many other React starters to be overly abstracted and hard to follow. A custom hooks library seems like it has this risk. So I'd like to see fairly detailed plans before greenlighting further action for the React template. 

Same questions for Angular & Vue, though I personally am less familiar with idioms in these communities.

**Comment by chadoh** [comment]
I agree that in its current form, near-api-js is not that pleasant to work with. But good news! The NEAR maintainers plan to improve the initialization & workflow of near-api-js which should lessen the need for your custom hook library. I currently expect these improvements to begin in Q2 2021.

**Comment by mehtaphysical** [comment]
Hey! Thanks for taking the time to consider this. I thought about this a bit and I think my ideas are two fold:

1. the use of global (window) variables isn't seen in many React applications.
2. because all the near integration is on the window object, IDEs can't help with code discovery (autocomplete).

That being said, I think your concern is very valid. I think the hooks library should just configure the NEAR connection and then directly expose things returned by `near-api-js`. It should be a very thin layer that.

For example, here is a small example using the window approach:

```js
import React, { useEffect } from 'react';
import { login } from './utils';o

export default function App() {
  useEffect(() => {
    if(!window.walletConnection.isSignedIn()) login();
  }, []);

  if(!wallet.isSignedIn()) return null;
  
  return <h1>{window.accountId}</h1>;
}
```

compared to a small example using a hooks approach:

```js
import React, { useEffect } from 'react'
import { useNearWallet } from 'near-react-hooks';

export default function App() {
  const wallet = useNearWallet()

  useEffect(() => {
    if(!wallet.isSignedIn()) wallet.requestSignIn();
  }, []);

  if(!wallet.isSignedIn()) return null;
  
  return <h1>{wallet.getAccountId()}</h1>;
}
```

The major advantages with the hooks approach are:

1. you import `useNearWallet` so you get a sense where the wallet is coming from. 
2. because `useNearWallet` is typed you get code completion.

**Comment by mehtaphysical** [comment]
Nice! ok i see the vision now. Seems legit to me!
